src/export/copy_across_regions/copy_snapshot_to_s3_export_supportable_region_function.py

```diff
@@ -0,0 +1,42 @@
+import os
+import boto3
+import constants
+import utility as util
+
+
+def lambda_copy_snapshot_to_s3_export_supportable_region(event, context):
+    """start export task of RDS snapshot to S3 bucket"""
+    region = os.environ['Region']
+    instance_id = event['identifier']
+    snapshot_id = instance_id + constants.SNAPSHOT_POSTFIX
+    snapshot_arn = util.get_instance_snapshot_arn(snapshot_id, region)
+    export_snapshot_supported_region = os.environ['ExportSnapshotSupportedRegion']
+    util.eval_export_exception(export_snapshot_supported_region)
+    export_snapshot_supported_region_rds = boto3.client('rds', export_snapshot_supported_region)
```

Comment:
can the region check and rds client creation be moved to the copy_db_snapshot method instead of passing many parameters

Reply:
@namitad unfortunately not. Actually the rds.copy_db_snapshot should be invoked within the parent method (only the parent method should exist). But If I leave it there there is no way to mock it(only at runtime you know the response type and its parameters of the boto3 call), so I can't test anything that happens after the rds.copy_db_snapshot method is invoked. Hence the additional dedicated method. If I follow your logic, even if I place all the statements (to reduce the parameters), we will end up with the same problem of mocking (there will be no tests). So if you can show me how we can do find out the runtime response type within the tests, I would happily do it. Otherwise lets leave it like this as UTs need beat parameter inconvenience.

Reply:
@despot i assume by response type and runtime parameters for boto3, you mean the response for the rds.copy_db_snapshot calls. these can be mocked as part of UTs. For example: https://github.com/intuit/Trapheus/blob/master/tests/unit/text_export_snapshot_s3_function.py#L25 
wouldnt this solve the issue of having another method and remove the parameter inconvenience?
Let me know if my understanding is correct.

Reply:
@namitad yes that is what I meant. Thanks for the direction. I tried [it (created a branch issue-4-parameters-fix-attempt](https://github.com/despot/Trapheus/tree/issue-4-parameters-fix-attempt) with [the new test](https://github.com/despot/Trapheus/blob/issue-4-parameters-fix-attempt/tests/unit/non_mock_common/test_copy_snapshot_to_s3_export_supportable_region_function.py) and [main code](https://github.com/despot/Trapheus/blob/issue-4-parameters-fix-attempt/src/export/copy_across_regions/copy_snapshot_to_s3_export_supportable_region_function.py)). 
, and I got:
`    mock_export_snapshot_supported_region_rds.copy_db_snapshot.return_value = self.mocked_copy_db_snapshot_response_good
AttributeError: 'str' object has no attribute 'copy_db_snapshot'`

Feel free to amend it or let me know what exactly is the problem. Since I follow the other test code by the letter and it seems like the test doesn't treat it as mock but like a string. Otherwise, let's leave it as is?